server/src/main/java/org/opensearch/throttling/tracker/NodePerformanceTracker.java

```diff
@@ -0,0 +1,194 @@
+        "node.global_cpu_usage.polling_interval",
+        TimeValue.timeValueMillis(500),
+        Setting.Property.NodeScope
+    );
+    public static final Setting<TimeValue> GLOBAL_CPU_USAGE_AC_WINDOW_DURATION_SETTING = Setting.positiveTimeSetting(
+        "node.global_cpu_usage.window_duration",
+        TimeValue.timeValueSeconds(30),
+        Setting.Property.Dynamic,
+        Setting.Property.NodeScope
+    );
+
+    public static final Setting<TimeValue> GLOBAL_JVM_USAGE_AC_POLLING_INTERVAL_SETTING = Setting.positiveTimeSetting(
+        "node.global_jvmmp.polling_interval",
+        TimeValue.timeValueMillis(500),
+        Setting.Property.NodeScope
+    );
+
+    public static final Setting<TimeValue> GLOBAL_JVM_USAGE_AC_WINDOW_DURATION_SETTING = Setting.positiveTimeSetting(
+        "node.global_jvmmp.window_duration",
+        TimeValue.timeValueSeconds(30),
+        Setting.Property.Dynamic,
+        Setting.Property.NodeScope
+    );
+
+    public NodePerformanceTracker(
+        PerformanceCollectorService performanceCollectorService,
+        ThreadPool threadPool,
+        Settings settings,
+        ClusterSettings clusterSettings
+    ) {
+        this.performanceCollectorService = performanceCollectorService;
+        this.threadPool = threadPool;
+        this.settings = settings;
+        this.clusterSettings = clusterSettings;
+        interval = new TimeValue(REFRESH_INTERVAL_MILLIS.get(settings));
+        initialize();
+    }
+
+    private double getAverageCpuUsed() {
+        return cpuUsageTracker.getAverage();
+    }
+
+    private double getAverageMemoryUsed() {
+        return memoryUsageTracker.getAverage();
+    }
+
+    private double getAverageIoUsed() {
+        // IO utilization percentage - this will be completed after we enhance FS stats
+        return 0;
+    }
+
+    public TimeValue getCpuWindow() {
+        return GLOBAL_CPU_USAGE_AC_POLLING_INTERVAL_SETTING.get(settings);
+    }
+
+    private void setCpuPercentUsed(double cpuPercentUsed) {
+        this.cpuPercentUsed = cpuPercentUsed;
+    }
+
+    private void setMemoryPercentUsed(double memoryPercentUsed) {
+        this.memoryPercentUsed = memoryPercentUsed;
+    }
+
+    private void setIoPercentUsed(double ioPercentUsed) {
+        this.ioPercentUsed = ioPercentUsed;
+    }
+
+    public double getCpuPercentUsed() {
+        return cpuPercentUsed;
+    }
+
+    public double getIoPercentUsed() {
+        return ioPercentUsed;
+    }
+
+    public double getMemoryPercentUsed() {
+        return memoryPercentUsed;
+    }
+
+    void doRun() {
+        setCpuPercentUsed(getAverageCpuUsed());
+        setMemoryPercentUsed(getAverageMemoryUsed());
+        setIoPercentUsed(getAverageIoUsed());
+        performanceCollectorService.addNodePerfStatistics(
+            LOCAL_NODE,
+            getCpuPercentUsed(),
+            getIoPercentUsed(),
+            getMemoryPercentUsed(),
+            System.currentTimeMillis()
```

Comment:
https://docs.oracle.com/javase/8/docs/api/java/lang/System.html#nanoTime -- this can be used  only when start time and end time is computed in same JVM. 

For different JVMs , we can't use this.